File: jpa/src/main/java/com/example/jpa/db/UserRepository.java
package com.example.jpa.db;

import org.springframework.data.jpa.repository.JpaRepository;


// Jpa 정의 들어가서 구현체 찾아보면 SimpleRepository 있음 이게 memoryDB 에서 만든 SimpleRepository 같은건데 JPA 에서 미리 만들어둠
public interface UserRepository extends JpaRepository<UserEntity, Long> {
}

File: service-security-kotlin-msa/db/src/main/kotlin/org/delivery/db/user/UserRepository.kt
package org.delivery.db.user

import org.delivery.db.user.enums.UserStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<UserEntity, Long> {

    fun findFirstByIdAndStatusOrderByIdDesc(userId: Long?, status: UserStatus?): Optional<UserEntity?>?

    fun findFirstByEmailAndPasswordAndStatusOrderByIdDesc(
        email: String?,
        password: String?,
        status: UserStatus?
    ): Optional<UserEntity?>?
}
